packages/wrapped-keys/src/lib/litActions.ts

```diff
@@ -1,146 +1,193 @@
-                Lit.Actions.setResponse({ response: errorMessage });
-                return;
-            }
+  if (!unsignedTransaction.chainId) {
+    Lit.Actions.setResponse({
+      response: 'Error: Missing required field: chainId',
+    });
+    return;
+  }
+
+  const decryptedPrivateKey = await Lit.Actions.decryptToSingleNode({
+    accessControlConditions,
+    ciphertext,
+    dataToEncryptHash,
+    chain: 'ethereum',
+    authSig: null,
+  });
+
+  console.log('decryptedPrivateKey');
+  console.log(decryptedPrivateKey);
+
+  if (!decryptedPrivateKey) {
+    // Exit the nodes which don't have the decryptedData
+    return;
+  }
+
+  const privateKey = decryptedPrivateKey.startsWith(LIT_PREFIX)
+    ? decryptedPrivateKey.slice(LIT_PREFIX.length)
+    : decryptedPrivateKey;
+  const wallet = new ethers.Wallet(privateKey);
+
+  console.log('unsignedTransaction.chain', unsignedTransaction.chain);
+  console.log('pkpAddress', pkpAddress);
+  const nonce = await Lit.Actions.getLatestNonce({
+    address: wallet.address,
+    chain: unsignedTransaction.chain,
+  });
+  console.log('nonce');
+  console.log(nonce);
+
+  const tx = {
+    to: unsignedTransaction.toAddress,
+    value: ethers.utils.hexlify(
+      ethers.utils.parseEther(unsignedTransaction.value)
+    ),
+    chainId: unsignedTransaction.chainId,
+    data: unsignedTransaction.dataHex,
+    nonce,
+  };
+
+  if (unsignedTransaction.gasPrice) {
+    tx.gasPrice = ethers.utils.parseUnits(unsignedTransaction.gasPrice, 'gwei');
+  } else {
+    tx.gasPrice = await Lit.Actions.runOnce(
+      { waitForResponse: true, name: 'gasPrice' },
+      async () => {
+        const rpcUrl = await Lit.Actions.getRpcUrl({
+          chain: unsignedTransaction.chain,
+        });
+        const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+        try {
+          const gasPrice = await provider.getGasPrice();
+          return ethers.utils.hexlify(gasPrice);
+        } catch (err) {
+          const errorMessage = 'Error: When getting gas price- ' + err.message;
+          Lit.Actions.setResponse({ response: errorMessage });
+          return;
         }
+      }
     );
+  }
+
+  if (unsignedTransaction.gasLimit) {
+    tx.gasLimit = unsignedTransaction.gasLimit;
+  } else {
+    tx.gasLimit = await Lit.Actions.runOnce(
+      { waitForResponse: true, name: 'gasLimit' },
+      async () => {
+        const rpcUrl = await Lit.Actions.getRpcUrl({
+          chain: unsignedTransaction.chain,
+        });
+        const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+
+        try {
+          const gasLimit = await provider.estimateGas(tx);
+          return ethers.utils.hexlify(gasLimit);
+        } catch (err) {
+          const errorMessage = 'Error: When estimating gas- ' + err.message;
+          Lit.Actions.setResponse({ response: errorMessage });
+          return;
+        }
+      }
+    );
+  }
```

Comment:
Same as above no need to use `runOnce` as at this point it's only one node executing it